File: 1003-check-if-word-is-valid-after-substitutions/1003-check-if-word-is-valid-after-substitutions.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const stack = []
    for (const c of s) {
        if (stack.length >= 2 && stack.at(-2) === "a" && stack.at(-1) === "b" && c === "c") {
            stack.pop()
            stack.pop()
        } else {
            stack.push(c)
        }
    }
    return !stack.length
};

File: 1003-check-if-word-is-valid-after-substitutions/1003-check-if-word-is-valid-after-substitutions.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for c in s:
            if len(stack) >= 2 and stack[-2] == 'a' and stack[-1] == 'b' and c == 'c':
                stack.pop()
                stack.pop()
            else:
                stack.append(c)
        return not stack